docs/grpc_disruption/interceptor_algorithm.md

```diff
@@ -0,0 +1,180 @@
+    0   -> Override: {}
+    1   -> Override: {}
+    2   -> Override: {}
+    3   -> Override: {}
+    4   -> Override: {}
+    5   -> Override: {}
+    6   -> Override: {}
+    ..  -> Override: ..
+    22  -> Override: {}
+    23  -> Override: {}
+    24  -> Override: {}
+    25  -> Error: NOT_FOUND
+    26  -> Error: NOT_FOUND
+    26  -> Error: NOT_FOUND
+    ..  -> Error: ...
+    61  -> Error: NOT_FOUND
+    62  -> Error: NOT_FOUND
+    63  -> Error: PERMISSION_DENIED
+    64  -> Error: PERMISSION_DENIED
+    65  -> Error: PERMISSION_DENIED
+    ..  -> Error: ...
+    99  -> Error: PERMISSION_DENIED
+    100 -> Error: PERMISSION_DENIED
+}
+```
+
+You cannot specify query percentages for a single endpoint which sum to over `100%`.
+
+### Simpler case of undefined query percentages
+
+You may have noted that the second example appears a tad complex. The intuition behind this design is to support the case where a user wants to quickly define a disruption which errors on all queries (replicating a bad roll out). For one error, the algorithm returns an error every time. For two errors, the algorithm returns half of the queries with the first error and half with the other.
+
+```
+spec:
+  grpc:
+    endpoints:
+      - endpoint: /chaos_dogfood.ChaosDogfood/order
+        error: NOT_FOUND
+      - endpoint: /chaos_dogfood.ChaosDogfood/order
+        error: PERMISSION_DENIED
+```
+
+Rather than constraining the user in how they mix and match this simple configuration style with the explicit `spec.gprc.endpoints[x].queryPercent` field, the current implementation would simply do its best to apply of the configurations.
+
+```
+PercentToAlteration {
+    0   -> Override: {}
+    1   -> Override: {}
+    2   -> Override: {}
+    3   -> Override: {}
+    4   -> Override: {}
+    5   -> Override: {}
+    6   -> Override: {}
+    ..  -> Override: ..
+    22  -> Override: {}
+    23  -> Override: {}
+    24  -> Override: {}
+    25  -> Error: NOT_FOUND
+    26  -> Error: NOT_FOUND
+    26  -> Error: NOT_FOUND
+    ..  -> Error: ...
+    61  -> Error: NOT_FOUND
+    62  -> Error: NOT_FOUND
+    63  -> Error: PERMISSION_DENIED
+    64  -> Error: PERMISSION_DENIED
+    65  -> Error: PERMISSION_DENIED
+    ..  -> Error: ...
+    99  -> Error: PERMISSION_DENIED
+    100 -> Error: PERMISSION_DENIED
+}
+```
+
+## Design Implications
+
+### Setting 0 as query percentage
+
+Users should avoid explicitly setting `query_pct: 0`. Because the Kubebuilder infers 0 for omitted `int`s, this configuration is synonymous for the chaos-controller to not setting a `query_pct` at all. This means that the alteration with this `query_pct` setting would actually be applied to the remaining queries that have not yet be claimed by another alteration. There isn't a straight forward way to chaos-controller to discriminate an unset `query_pct` from one explicitly set to `0`, so users may be surprised by inputting the value `0` and receving many of errors with that input.
```

Comment:
```suggestion
Users should avoid explicitly setting `query_pct: 0`. Because the Kubebuilder infers 0 for omitted `int`s, this configuration is synonymous for the chaos-controller to not setting a `query_pct` at all. This means that the alteration with this `query_pct` setting would actually be applied to the remaining queries that have not yet be claimed by another alteration. There isn't a straight forward way to chaos-controller to discriminate an unset `query_pct` from one explicitly set to `0`, so users may be surprised by inputting the value `0` and receiving many of errors with that input.
```

"many of errors" could flow more clearly.

What are we hoping to warn people against here? I don't know why anyone would ever explicitly set a query percentage of 0 and think it would never happen, because why even add that alteration then? What's here is perfectly fine, but we can probably also just tell people that `0` is a special value and should expect it to be treated that way